### ISSUE TITLE
debug validator for ieeg (bep010): order of columns in channels.tsv

### ISSUE THREAD

**Issue by irisgroen** [bids-standard/legacy-validator#666]
While checking an bids-example dataset (https://github.com/bids-standard/bids-examples/pull/138) for the ieeg (bep010) branch I came across this error, which I don't know how to address because to me the order of columns under 'Evidence' seems consistent with the instructions. Could it be due to there being more columns than the required ones?

	4: [ERR] The column names of the channels file must begin with ['name', 'type', 'units', 'sampling_frequency', 'low_cutoff', 'high_cutoff', 'notch', 'reference'] (code: 72 - MISSING_TSV_COLUMN_IEEG_CHANNELS)
		./sub-som682/ses-nyuecog01/ieeg/sub-som682_ses-nyuecog01_task-hrfpattern_run-01_channels.tsv
			@ line: 1 character: 16
			Evidence: name,type,units,sampling_frequency,low_cutoff,high_cutoff,notch,reference,group,description,status,status_description

@choldgraf @dorahermes

**Comment by sappelhoff** [comment]
yes, this is due to the validator lagging behind the specification. @choldgraf has started to work on this in #652 

see also the conversation in https://github.com/bids-standard/bids-examples/pull/133

for related problems. Hopefully, these will be resolved very soon. Thanks for reporting!

**Comment by dorahermes** [comment]
lgtm

**Comment by sappelhoff** [comment]
Hi @irisgroen has your problem been solved through #709 ?

**Comment by irisgroen** [comment]
> Hi @irisgroen has your problem been solved through #709 ?

Unfortunately I still get the same error:
`	3: [ERR] The column names of the channels file must begin with ['name', 'type', 'units', 'sampling_frequency', 'low_cutoff', 'high_cutoff', 'notch', 'reference'] (code: 72 - MISSING_TSV_COLUMN_IEEG_CHANNELS)
		./sub-som682/ses-somecog01/ieeg/sub-som682_ses-somecog01_task-spatialobject_run-01_channels.tsv
			@ line: 1 character: 16
			Evidence: name,type,units,sampling_frequency,low_cutoff,high_cutoff,notch,reference,group,description,status,status_description` 

I've looked at the  _channels.tsv file again but I can't see what the problem is with character 16 on line 1 (this would be the word 'sampling_frequency' I think). I attached the first file here if that helps:
[sub-som682_ses-somecog01_task-spatialobject_run-01_channels.tsv.zip](https://github.com/bids-standard/bids-validator/files/2901262/sub-som682_ses-somecog01_task-spatialobject_run-01_channels.tsv.zip)

**Comment by sappelhoff** [comment]
Which version of the validator are you using? The error message seems to be the old one. The new one should be as described here:

https://github.com/bids-standard/bids-validator/blob/b2f0dd3e6a37eb6caa2df0e0b16f535942636536/bids-validator/utils/issues/list.js#L399-L404

Soon (in a few hours to days) We'll also be able to run our iEEG datasets against the browser version of the validator, so that might help us to debug your problem as well (less chance of messing up with one's own installation of the validator)

**Comment by irisgroen** [comment]
Thanks - from the 'reason' field in the linked code block I understand now that the required columns should no longer include sampling_frequency. When I moved that column the validator no longer gave the error! I updated the channels.tsv files in my pull request in bids_examples accordingly (https://github.com/bids-standard/bids-examples/pull/138).

I am not sure about which version I am using now however. I did a pull request just prior to testing and thought that meant I was using the most recent version. But if I type:
`bids-validator --version`
It says I am using version 1.1.1, which is not the most recent release (1.1.5). How do I obtain that version? I tried removing and cloning the bids-validator repo again. I also tried listing all the tags and switching to the latest one using
`git checkout 1.1.5`
But I still get the same bids-validator version and also the same (old) error message if I test the code on the old channels.tsv file (which still included sampling_frequency in the list of required columns).

Thank you!

**Comment by sappelhoff** [comment]
> When I moved that column the validator no longer gave the error!

Yay!

> I am not  sure about which version I am using now however.

mmmh :confused: there are a couple of things here:

- when cloning the bids-validator and then installing the development version from the master branch using `yarn`, you should get version `0.0.0` which is a known bug that has not yet been fixed: #602 

- when instead using `npm install -g bids-validator` I personally get version `1.1.5`, as expected.

- I am at a loss how
    - you can get a version at all from the development install (see #602 again)
    - that version is somewhere in the middle ... not latest and not super old

What operating system are you on?

I install bids-validator (Ubuntu 18.04) as such:

1. install yarn as described [here](https://linuxize.com/post/how-to-install-yarn-on-ubuntu-18-04/)
2. `git clone https://github.com/bids-standard/bids-validator`
3. `cd bids-validator`
4. `yarn`
5. Then within the `bids-validator` directory: `cd bids-validator/bin`
6. Finally, use it via `./bids-validator --version` --> yields `0.0.0`

**Comment by irisgroen** [comment]
I am using OS X 10.12.6. I originally installed bids-validator using `npm install` and then I updated it using `git pull` inside the bids-validator repo. Perhaps that is how I got stuck at 1.1.1.? In any case, I just re-installed the validator using npm and I now get version 1.1.5, and the 'new' error message 72. So we can close this issue! (Unfortunately a few new things popped up, but I think some of those are addressed in other issues, will check those first before posting anything new). Thanks for your help!
